### DISCUSSION TITLE
require-dev in composer.lock

### DISCUSSION THREAD

**Discussion by b1rdex** [composer/composer#11075] [General]
Hi. I'm wondering – why `composer.lock` contains all the `require-dev` statements for dependencies? Is there any use case for that?

**Comment by stof** [comment]
Those are indeed not needed, as they are not used by composer.

IIRC, the locker dumps all metadata provided by packagist, and `require-dev` was not stripped when creating the v2 metadata format.

**Comment by talkinnl** [comment]
Do you consider changing that in the future? Maybe add the idea to a new minor (major?) version?

This change would really cut down on the diffs made by updates, which also helps if you ever want to double-check what bots like Renovate and Dependabot did.
Probably is an environmentally friendly change too. ;)

**Reply by Seldaek** [reply]
If you want to review changes to the lock file I'd highly recommend a packagist.com subscription, the [Update Review](https://packagist.com/docs/update-review) feature makes this much more usable than visually parsing json diffs.. This file really wasn't meant for human consumption and I don't think it's worth optimizing this much but yes technically we don't really need the require-dev for dependencies there.

**Reply by MrSuddenJoy** [reply]
@Seldaek OK, granted. Question still stands though: **why** `require-dev` block/section (call it whatever) is still there if its no longer used?

**Reply by Seldaek** [reply]
Well, we include all the package metadata because you can then use it with composer show --locked for example to see all details.

**Comment by ProtocolLive** [comment]
My server takes too more time to process the require-dev, even with --no-dev...
